### ISSUE TITLE
Would you be willing to write an exporter?

### ISSUE THREAD

**Issue by ReindeerFloatilla** [rfsheffer/nod_nad_to_fbx#2]
Hi there. There is now a .nod/nad Noesis plugin that has been created. I've uploaded it here.

https://www.moddb.com/games/vampire-the-masquerade-redemption/downloads/nod-noesis-plugin

It currently imports/exports .nod and imports .nad. However it does not export. Would you be willing to write an exporter for .nad? It would greatly benefit the modding community as we could now create custom animations for the game.

**Comment by rfsheffer** [comment]
Hey!
You seem to have figured the whole thing out since you were able to import the NAD files. Exporting them back is just reversing the process unless I am missing something. I remember having trouble with the inverse binding pose matrix that was stored in the data and not being able to figure out exactly how it was structured. I assume you figured it out with your importer.
So your goal is to get animations back into VtMR I assume. So the best process here would be to get the data into Blender which would be great for editing and be able to export it back to the game's native formats. So some tool for Blender would be very ideal. I am happy to hear there is still some momentum around VtMR modding. It is always in the back of my mind and wouldn't mind helping out but I keep thinking about how great it would be to bring VtMR to a modern game engine like Unreal. Build some map converter, then all the assets like meshes, textures and FX, then the gameplay code which is in java?
Anyway, I can dream.
Let me know what is keeping you from exporting back to NAD and I will see what I can do to help.
Cheers!

**Comment by ReindeerFloatilla** [comment]
I'm not the programmer of the importer plugin it was someone who is currently uninterested in finishing it. I'm fairly certain the importer figured out the inverse binding pose matrix. 

Right now we use Noesis in conjunction with Blender. I wrote a tutorial for doing it with .nod we just need it for .nad. I would love some kind of simpler exporter solution but for now this is at least better than using Milkshape.
https://www.moddb.com/games/vampire-the-masquerade-redemption/tutorials/how-to-import-models-from-blender-into-vampire-the-masquerade-redemption

I'd rather stick to the Nod engine as opposed to working towards porting VtmR to Unreal. I want to create a custom singleplayer chronicle for the game but would need custom animations for some of the more ambitious ideas I have.

**Comment by rfsheffer** [comment]
OK so my thinking is you guys would be better off with a blender plugin that can import and export NOD and NAD data.
I will deprecate this project "nod_nad_to_fbx" and start a new repo soon for the blender plugin. Wish me luck in getting this all working properly, I already have the models importing into blender directly using a custom plugin.
![Screenshot 2023-11-12 143010](https://github.com/rfsheffer/nod_nad_to_fbx/assets/1026377/755dbe3c-d8ec-48cc-ad9e-f84eeacf3ac2)

**Comment by ReindeerFloatilla** [comment]
This is so exciting. If you can get exporting working I would be forever in your debt. I would also write a tutorial on how to do both meshes and animations

**Comment by ReindeerFloatilla** [comment]
If you want to easily test I recommend replacing Christof.nod and Christof's idle animation chris_stand_util. Place them in vampire the masquerade redemption\3D\models and \anims respectively. Then go to multiplayer and select LAN and you should see the replaced model/animation over Dark Ages Christof.

The engine is similar to Quake in that you can place files in the same directory structure as the .nob files in the root of the folder. The game will load them first. You can also open .nob with 7zip.

**Comment by rfsheffer** [comment]
Thanks for the tips, you will have to be patient with me while I work on this. Hopefully, it isn't too complicated.

**Comment by rfsheffer** [comment]
Progress report:
Way further than I was before. I have the skeleton in and bound to the mesh. I will be starting the process of importing NAD animations next. Once I am happy with all that I can start exporting back to NOD / NAD.

I do have a question tho:
Where in the VtMR modding community hanging out these days? Are people still playing and making multiplayer campaigns like the old days?

**Comment by ReindeerFloatilla** [comment]
Are you a Tremere? Because you are an absolute magician!

Right now we hang out on the PlanetVampire discord. Planetvampire as a website went down earlier this year. There's a successor site called PlanetVampyre with a new forum that we've been using. There's also talk about making a new VtmR wiki. 

https://clinchnet.com/planetvampyre/
https://discord.com/invite/kCNFRzabAk

I've been personally working on preserving and archiving lost VtmR mods since PlanetVampire went down. I've managed to upload hundreds of stuff to moddb. I've also been trying to make my own custom player skins. There are some creators currently working on making custom multiplayer maps as well. I hope to make a custom singleplayer chronicle one day.

**Comment by ReindeerFloatilla** [comment]
Hi there. Are you still working on the project? Do please let me know at your earliest convenience. We're just hoping to get to work on importing some animations into the game.

We're also currently working on a new forum as the host of our previous one was unreliable. Will post a link here eventually.

**Comment by rfsheffer** [comment]
Yep, still working on it. My next stage is to import the NAD data. Once that works then I just have to reverse the process to create an exporter. I haven't worked on it for a couple of weeks because I am really busy with my job, closing out projects for the year. I hope to get some time to work on it this week.

Cheers!

**Comment by ReindeerFloatilla** [comment]
Nice to hear. Can't wait. I have so many ideas for random poses for Christof.

**Comment by ReindeerFloatilla** [comment]
Hey. Just checking in are you still working on this project?

**Comment by robsonner** [comment]
Hello @rfsheffer , I've seen that you were working on VTM Redemption models import/export for Blender. I can't find the new repo you mentioned in this thread. Is it possible that the source code will be publicly available? I’ve actually started working on a similar project (importing NOD models to Blender), but I hit a wall. I wonder if you had similar problems and maybe resolved them.

My problem is that when I create the skeleton from NOD data, I can use rest_translate to get the skeleton into a T-pose. However, the mesh in most of the models is not in a T-pose. Did you manage to generate a proper skeleton in Blender that matches the mesh position?

Here is my result:

<img width="608" height="514" alt="Image" src="https://github.com/user-attachments/assets/552b555a-786f-4b08-8aaa-594680b81f09" />

**Comment by ReindeerFloatilla** [comment]
Hi @robsonner the nod/nad noesis plugin now does full import/export. 
https://www.moddb.com/games/vampire-the-masquerade-redemption/tutorials/how-to-import-models-from-blender-into-vampire-the-masquerade-redemption
https://www.moddb.com/games/vampire-the-masquerade-redemption/tutorials/how-to-import-animations-from-blender-into-vampire-the-masquerade-redemption

**Comment by robsonner** [comment]
Hi @ReindeerFloatilla I am aware of that :P I even tried to analyze the code of Noesis plugin to get the info on how to do the skeleton creation properly but the trail ends at the Noesis specific code that I have no access to. I wanted to skip a middle man and just have a working blender solution :)

**Comment by rfsheffer** [comment]
Hey @robsonner 

Sorry for the really late reply. Here is the latest code for the NOD NAD plugin I was working on for blender. It was for Blender 3.6 so will probably need fixes to work, but you can see how I solved the reference pose problem.

Cheers!

NOTE: The NAD functionality is not operational at all. I actually fed the problem into AI recently as a test and it thought it could get it all working with blender import and export so maybe that is a path forward. :)

[io_scene_nod.zip](https://github.com/user-attachments/files/25460518/io_scene_nod.zip)
